aot/src/main/java/com/dylibso/chicory/experimental/aot/TypeStack.java

```diff
@@ -34,14 +34,14 @@ public void push(ValueType type) {
 
     public void pop(ValueType expected) {
         var actual = types().pop();
-        if (expected != actual) {
+        if (!expected.equals(actual)) {
```

Comment:
Do you have a "rule of thumb" to follow on when comparing `opcodes` only and when using `equals`?

Reply:
I think as a rule of thumb, use `equals` now that ValueType is an object (this is equivalent to comparing ValueType.id()). I'm not sure of a case where we only want to compare opcodes, possibly when the code looks like:

```
private Class javaClassFromValueType(ValueType type) {
        switch (type.opcode()) {
            case I32:
                return int.class;
            case I64:
                ....
```

so enumerating based on the type

Reply:
looks good :+1: